kt/entry-generation/godot-entry-generator/src/main/kotlin/godot/entrygenerator/generator/hintstring/PropertyHintStringGeneratorProvider.kt

```diff
@@ -0,0 +1,55 @@
+package godot.entrygenerator.generator.hintstring
+
+import godot.entrygenerator.ext.hasAnnotation
+import godot.entrygenerator.ext.isCompatibleList
+import godot.entrygenerator.ext.isCoreType
+import godot.entrygenerator.ext.isReference
+import godot.entrygenerator.model.IntFlagHintAnnotation
+import godot.entrygenerator.model.MultilineTextHintAnnotation
+import godot.entrygenerator.model.PlaceHolderTextHintAnnotation
+import godot.entrygenerator.model.RegisteredProperty
+import godot.entrygenerator.model.TypeKind
+
+
+object PropertyHintStringGeneratorProvider {
+
+    fun provide(
+        registeredProperty: RegisteredProperty
+    ): PropertyHintStringGenerator {
+        return when {
```

Comment:
Wouldn't that be simpler for future hint string additions to iterate on the list of annotation and do the 'when' on the annotation class itself instead of the class of the property?
That way we can manage the when blocks on a per annotation basis so we don't have the same annotation repeated several times across the file.

Reply:
Hmm not sure if possible. For some hint strings i need information about the property itself not only the annotation.
For example for the `ArrayHintStringGenerator`: https://github.com/utopia-rise/godot-kotlin-jvm/blob/37914fc35e6649c08f6dec120f06296a2c33a99a/kt/entry-generation/godot-kotlin-entry-generator/src/main/kotlin/godot/entrygenerator/generator/property/hintstring/ArrayHintStringGenerator.kt#L11

Reply:
Ho you would still have the same data available I was just talking about how you currently select the right genererator:
-Check the type of the property
-Check if  it owns one of the annotations
-If yes return the right generator

I was proposing the reverse:
- Check if  it owns one of the annotations
- Check the type.
- If yes, return the right generator.

It's just so everything related to one particular annotation is handled in the same place.

Reply:
ah. Now i understand. Yeah that makes more sense. I'll look into it.

Reply:
done. feel free to unresolve if what i've done is not what you expected